src/query/expression/src/type_check.rs

```diff
@@ -517,6 +517,7 @@ pub fn can_auto_cast_to(
                 .all(|(src_ty, dest_ty)| can_auto_cast_to(src_ty, dest_ty, auto_cast_rules))
         }
         (DataType::Number(n), DataType::Decimal(_)) if !n.is_float() => true,
+        (DataType::String, DataType::Decimal(_)) => true,
```

Comment:
I think this rule should be forbidden.  We will not allow string auto-cast to decimal, even string to float is not allowed.

Reply:
ok

Reply:
fixed